### PR TITLE
{doc} Add `Scenario Test Best Practice` section in `authoring_tests`

### PR DESCRIPTION
**Pull request by kairu-ms**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by yonzhan — COMMENTED** [review]
If all combinations of parameters are covered, the existing examples are automatically covered, right?

**Comment by yonzhan** [comment]
This is a good practice to incorporate boundary condition in test cases.

**Review by yonzhan — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

File: doc/authoring_tests.md [hunk 1]
```diff
@@ -36,6 +36,13 @@ It is a requirement for the command owner to maintain their test in live mode.
 * Name test methods in the following format: `test_<module>_<feature>`.
 * The scenario test must be able to run repeatedly in live mode. The feature owner is responsible of maintaining their scenario tests.
 
+### Scenario Test Best Practice
+
+* 100% commands coverage of module: Scenerio tests __MUST__ coverage all commands in the module except the `wait` commands. Please use `azdev cmdcov {module}` to check command coverage of module.
```
Comment (kairu-ms) [inline doc/authoring_tests.md:41]:
```suggestion
* 100% commands coverage of module: Scenerio tests __MUST__ cover all commands in the module except the `wait` commands. Please use `azdev cmdcov {module}` to check command coverage of module.
```

File: doc/authoring_tests.md [hunk 2]
```diff
@@ -36,6 +36,13 @@ It is a requirement for the command owner to maintain their test in live mode.
 * Name test methods in the following format: `test_<module>_<feature>`.
 * The scenario test must be able to run repeatedly in live mode. The feature owner is responsible of maintaining their scenario tests.
 
+### Scenario Test Best Practice
+
+* 100% commands coverage of module: Scenerio tests __MUST__ cover all commands in the module except the `wait` commands. Please use `azdev cmdcov {module}` to check command coverage of module.
```
Comment (wangzelin007) [inline doc/authoring_tests.md:41]:
```suggestion
* 100% commands coverage of module: Scenerio tests __MUST__ cover all commands in the module except the `wait` commands. Please use `azdev cmdcov {module}` to check command coverage of module. If cannot add a test because of some reason, please add missing_command_test_coverage in the linter_exclusions.yml.
```

File: doc/authoring_tests.md [hunk 3]
```diff
@@ -36,6 +36,13 @@ It is a requirement for the command owner to maintain their test in live mode.
...
+* 100% examples coverage of command: Scenerio tests __MUST__ coverage all examples of the command.
+* 100% arguments coverage of command: Scenerio tests should coverage all arguments in the command. Please use `azdev cmdcov {module} --level argument` to check argument coverage.
```
Comment (wangzelin007) [inline doc/authoring_tests.md:43]:
```suggestion
* 100% arguments coverage of command: Scenerio tests should coverage all arguments in the command. Please use `azdev cmdcov {module} --level argument` to check argument coverage. If cannot add a test because of some reason, please add missing_parameter_test_coverage in the linter_exclusions.yml.
```

File: doc/authoring_tests.md [hunk 4]
```diff
@@ -36,6 +36,13 @@ It is a requirement for the command owner to maintain their test in live mode.
...
 
+### Scenario Test Best Practice
+
+* 100% commands coverage of module: Scenerio tests __MUST__ cover all commands in the module except the `wait` commands. Please use `azdev cmdcov {module}` to check command coverage of module. If cannot add a test because of some reason, please add missing_command_test_coverage in the linter_exclusions.yml.
+* 100% examples coverage of command: Scenerio tests __MUST__ coverage all examples of the command.
+* 100% arguments coverage of command: Scenerio tests should coverage all arguments in the command. Please use `azdev cmdcov {module} --level argument` to check argument coverage. If cannot add a test because of some reason, please add missing_parameter_test_coverage in the linter_exclusions.yml.
+* 100% boundary values coverage of argument: Scenerio tests should coverage boundary values of argument, especially ``(blank value), `null`, `0` and `False` values. These values have different meanings, but will all resolve to `False` in conditional expressions (`if {variable}`) of Python. It's important to make sure they work as expected.
```
Comment (jiasli) [inline doc/authoring_tests.md:44]: Empty content inside `` ` `` leads to rendering error:

image

Comment (jiasli) [reply doc/authoring_tests.md:44]:
```suggestion
* 100% boundary values coverage of argument: Scenerio tests should coverage boundary values of argument, especially `''` (empty string), `null`, `0` and `False` values. These values have different meanings, but will all resolve to `False` in conditional expressions (`if {variable}`) of Python. It's important to make sure they work as expected.
```

**Review by jiasli — APPROVED** [review]
